portfolio-project-matching-app/pages/browseProjects.js

```diff
@@ -17,41 +18,90 @@ const browseProjects = () => {
     // array of projects being hidden by filters
     const [hiddenProjects, setHiddenProjects] = useState([])
 
-    const getProjects = async () => {
+    // array of all technologies
+    const [allTechnologies, setAllTechnologies] = useState([])
+
+    // array of visible technology
+    const [visibleTechnologies, setVisibleTechnologies] = useState([])
+
+    // array of hidden technology objects
+    const [hiddenTechnologies, setHiddenTechnologies] = useState([])
+
+    const initializeProjects = async () => {
         const projects = await getAllProjects()
         setVisibleProjects(projects)
     }
 
+    const initializeTechnologies = async () => {
+        const technologies = await getAllTechnologies()
+        setAllTechnologies(technologies)
+    }
+
+    const onFilterClick = (choice) => {
+        // this should most likely be handled by FilterButtons
+        // hand work off to other functions based on whether add/remove
+        hiddenTechnologies.includes(choice) ? removeFilter(choice) : addFilter(choice)
+    }
+
+    const addFilter = (choice) => {
```

Comment:
I realize that a lot of what I did in `addFilter` and `removeFilter` looks messy and wordy. I want to get this looking more tidy once the deadline isn't breathing down our necks.